src/cbmc/README.md

```diff
@@ -76,13 +76,15 @@ by other mechanism may categorise their properties differently.
 
 ### Coverage mode
 
-There is one further BMC mode that differs more fundamentally: when `--cover` is
-passed, assertions in the program text are converted into assumptions (these
+There is one further BMC mode that differs more fundamentally: when `--cover`
+is passed, assertions in the program text are converted into assumptions (these
```

Comment:
I don't want to open a massive can of worms here but ... why is this the default?  Is this actually useful?  Given that there are options for `--no-assertions`, `--no-assumptions` and `--assert-to-assume` (whether they work or even do anything at all ... is an open question) could we make the transformation of assertions and assumptions orthogonal to `--cover`.  It feels like this should be possible and even desirable...

Reply:
@martin-cs I think this is the default because whoever wrote this had the expectation that programs should stop at failed assertions. I suppose we could change it, but I’m always a bit wary about changing defaults.

Reply:
@martin-cs FWIW because cover treats each failed goal as something relating to coverage right now we have to do _something_ with asserts. That said, I think the alternative (just making cover cleverer about how it counts coverage) would be more desirable.

Reply:
Yes, if you are doing a naive count then, I see the need to do something with pre-existing assertions or otherwise true assertions (generally a good thing) will count against your coverage.  Smarter counting would certainly be one way to address this.

It feels to me that making the "doing something about asserts" independent might be a desirable option as well.  I can see use-cases for converting to SKIP, to ASSUME and leaving as they are.

Then again, I am not a user of this code at the moment and do not have time to work on it so my views maybe not so important.